### PR TITLE
feat: add typescript support for vue

### LINKED ISSUE #2428: Type declarations used in components can't be imported
### Which generators are impacted?

- [ ] All
- [ ] HTML
- [ ] React
- [ ] Angular
- [X] Vue
- [ ] Web components
- [ ] Power Apps

### Reproduction case

For strong typing in our application we want to import and use types like _DefaultVariantType_ that are located in dist/shared.
At the moment an import is not possible.

### Expected Behaviour

These types should be exported too in the default type declaration file (dist/index.d.ts)

### Screenshots

_No response_

### Browser version

None

### Add any other context about the problem here.

_No response_

### LINKED ISSUE #2441: All props of the Vue components have the type any
### Which generators are impacted?

- [ ] All
- [ ] HTML
- [ ] React
- [ ] Angular
- [X] Vue
- [ ] Web components
- [ ] Power Apps

### Reproduction case

All props of the Vue components have the type any, which means that there's no auto-suggestions / -complete for props.

### Expected Behaviour

types would be provided.

### Screenshots

_No response_

### Browser version

None

### Add any other context about the problem here.

_No response_

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

closes #2428
closes #2441

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: output/vue/scripts/add-typescript.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		.map((part) => `${part[0].toUpperCase()}${part.slice(1)}`)
+		.join('');
+
+/**
+ * Props are generated like this: `readonly id?: any;`
+ * We replace it with the correct type like DBMyComponent["id"]
+ * @param input {string}
+ * @param component {string}
+ */
+const replaceAnyType = (input, component) => {
```
Comment (bruno-sch) [inline output/vue/scripts/add-typescript.js:18]: Maybe with plural `replaceAnyTypes`?

File: output/vue/scripts/add-typescript.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		fileContent = fileContent.replace(
+			propLine,
+			propLine.replace('any', `${propModel}["${prop}"]`)
+		);
+	}
+
+	return `import { ${propModel} } from "./model";\n\n${fileContent}`;
+};
+
+const addTypescript = () => {
```
Comment (bruno-sch) [inline output/vue/scripts/add-typescript.js:37]: Wouldn't `fixTypescript` or `addTypescriptSupport` make more sense?

File: output/vue/scripts/add-typescript.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { readdirSync } from 'node:fs';
+import Replace from 'replace-in-file';
+
+const distDir = './dist/components';
+
+const getComponentName = (componentName) =>
+	componentName
+		.split('-')
+		.map((part) => `${part[0].toUpperCase()}${part.slice(1)}`)
+		.join('');
```
[the remark this answers is not in the document]
Comment (nmerget) [reply output/vue/scripts/add-typescript.js]: fixed

File: output/vue/scripts/add-typescript.js (hunk 2 above)
Comment (nmerget) [reply output/vue/scripts/add-typescript.js]: fixed

File: output/vue/scripts/add-typescript.js (hunk 1 above)
Comment (nmerget) [reply output/vue/scripts/add-typescript.js]: fixed

**Review by bruno-sch — APPROVED** [review]

**Pull request merged**